gradle-enterprise-conventions-gradle-plugin/plugins/gradle-5-or-newer/src/main/java/com/myorg/GradleEnterpriseConventionsPlugin.java

```diff
@@ -0,0 +1,74 @@
+package com.myorg;
+
+import com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension;
+import com.gradle.enterprise.gradleplugin.GradleEnterprisePlugin;
+import com.gradle.scan.plugin.BuildScanExtension;
+import com.gradle.scan.plugin.BuildScanPlugin;
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.initialization.Settings;
+import org.gradle.caching.configuration.BuildCacheConfiguration;
+import org.gradle.caching.http.HttpBuildCache;
+import org.gradle.util.GradleVersion;
+
+public class GradleEnterpriseConventionsPlugin implements Plugin<Object> {
+
+    @Override
+    public void apply(Object target) {
+        if (target instanceof Settings) {
+            if (!isGradle6OrNewer()) {
+                throw new GradleException("For Gradle versions prior to 6.0, gradle-enterprise-conventions must be applied to the Root project");
+            }
+            applySettingsPlugin((Settings) target);
+        } else if (target instanceof Project) {
+            if (isGradle6OrNewer()) {
+                throw new GradleException("For Gradle versions 6.0 and newer, gradle-enterprise-conventions must be applied to Settings");
+            }
+            applyProjectPlugin((Project) target);
+        }
+    }
+
+    private void applySettingsPlugin(Settings settings) {
```

Comment:
Maybe include the Gradle version numbers in these applyMethods to make things a bit clearer.

Reply:
I renamed the methods, and I do think it's a bit clearer now.